include/ringmesh/mesh/mesh_builder.h

```diff
@@ -180,7 +194,22 @@ namespace RINGMesh
         void delete_vertices( const std::vector< bool >& to_delete )
         {
             do_delete_vertices( to_delete );
+            index_t new_size = mesh_base_.nb_vertices();
+            std::vector< index_t > permutation;
+            permutation.resize( new_size );
+            index_t i = 0;
+            for( auto d : range( to_delete.size() ) )
+            {
+                if( !to_delete[d] )
+                {
+                    permutation[i] = d;
+                    i++;
+                }
+            }
+            mesh_base_.vertex_attributes_manager_.apply_permutation(
+                permutation );

```

Comment:
All this new code block concerns attributes. Maybe you can create a function for these 10-15 lines and call this function here